pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -206,21 +206,44 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	return err
 }
 
-func (c *Reconciler) reconcile(ctx context.Context, route *v1alpha1.Route) error {
+func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 	logger := logging.FromContext(ctx)
-	route.Status.InitializeConditions()
+	r.Status.InitializeConditions()
 
-	logger.Infof("Reconciling route :%v", route)
-	logger.Info("Creating/Updating placeholder k8s services")
-	if err := c.reconcilePlaceholderService(ctx, route); err != nil {
+	logger.Infof("Reconciling route :%v", r)
+	// configure traffic based on the RouteSpec.
+	traffic, err := c.configureTraffic(ctx, r)
+	if traffic == nil || err != nil {
+		// Traffic targets aren't ready, no need to configure child resources.
 		return err
 	}
 
-	// Call configureTrafficAndUpdateRouteStatus, which also updates the Route.Status
-	// to contain the domain we will use for Gateway creation.
-	if _, err := c.configureTraffic(ctx, route); err != nil {
+	logger.Info("Updating targeted revisions.")
+	// In all cases we will add annotations to the referred targets.  This is so that when they become
+	// routable we can know (through a listener) and attempt traffic configuration again.
+	if err := c.reconcileTargetRevisions(ctx, traffic, r); err != nil {
+		logger.Errorf("Failed to update target revisions: %v", err)
+	}
+
+	// Update the information that makes us Targetable.
+	r.Status.Domain = routeDomain(ctx, r)
+	r.Status.DomainInternal = resourcenames.K8sServiceFullname(r)
+	r.Status.Targetable = &duckv1alpha1.Targetable{
```

Comment:
Should we set these before we've even created the placeholder service?

Reply:
ClusterIngress creation rely on this (Domain in route status).
In fact placeholder service is kind of out of control in route status: its status did not get reflected in route status. Route would still be ready even if failing to reconcile placeholder service.
Of course we can defer the status update to the end of the func, to reflect service unready status in route.
But I feel that kind of weird, as existing conditions are not indeed designed for placeholder service.
Maybe add extra condition for it, if we intend to do so.

Reply:
Could we just create the placeholder earlier then, if the ClusterIngress relies on it?

Reply:
Sorry, I didn't get your point.
ClusterIngress rely on the `Domain` in status here, and the placeholder rely on the ingress (main change in this PR):
Status.Domain > ClusterIngress > placeholder
So whose order would you suggest to adjust?